src/node-file-trace.ts

```diff
@@ -1,17 +1,42 @@
 import { NodeFileTraceOptions, NodeFileTraceResult, NodeFileTraceReasons, Stats, NodeFileTraceReasonType } from './types';
-import { basename, dirname, extname, relative, resolve, sep } from 'path';
+import { basename, dirname, extname, relative, resolve, sep, join } from 'path';
+import * as url from 'url';
 import fs from 'graceful-fs';
 import analyze, { AnalyzeResult } from './analyze';
 import resolveDependency, { NotFoundError } from './resolve-dependency';
 import { isMatch } from 'micromatch';
 import { sharedLibEmit } from './utils/sharedlib-emit';
-import { join } from 'path';
 import { Sema } from 'async-sema';
 
 const fsReadFile = fs.promises.readFile;
 const fsReadlink = fs.promises.readlink;
 const fsStat = fs.promises.stat;
 
+type ParseSpecifierResult = {
+  path: string;
+  queryString: string | null
+}
+
+// Splits an ESM specifier into path and querystring (including the leading `?`). (If the specifier is CJS,
+// it is passed through untouched.)
+export function parseSpecifier(specifier: string, cjsResolve: boolean = true): ParseSpecifierResult {
+  let path = specifier;
+  let queryString = null;
+
+  if (!cjsResolve) {
+    const specifierUrl = url.parse(specifier);
```

Comment:
I just learned `url.parse()` was deprecated (again) so lets use `new URL()` instead

- https://github.com/nodejs/node/pull/44919

Reply:
Any suggestion on how to deal with URL not being able to handle relative paths? Would it be ok for us to use some random base url (eg `example.com`)?

Reply:
I think this is a job for [`pathToFileURL()`](https://nodejs.org/api/url.html#urlpathtofileurlpath)